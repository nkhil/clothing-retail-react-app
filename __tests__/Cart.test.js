import "@babel/polyfill"
import React from "react"
import ShoppingCart from "../src/App/components/Cart"
import renderer from "react-test-renderer"
import ReactDOM from "react-dom"

require("react-router-dom")
jest.mock("react-router-dom")

const mockProducts = {
  product1: {
    productName: "Almond Toe Court Shoes, Patent Black",
    productCategory: "Women's Footwear",
    productPrice: 9900,
    productQuantity: 5,
  },
  product2: {
    productName: "Suede Shoes, Blue",
    productCategory: "Women's Footwear",
    productPrice: 4200,
    productQuantity: 4,
  },
}

const mockShoppingCart = { product1: 1, product2: 1 }

const mockRemoveFromCart = () => {}
const mockCalculateDiscountedTotal = () => 2300

test("ShoppingCart renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(
    <ShoppingCart
      products={mockProducts}
      shoppingCart={mockShoppingCart}
      removeFromCart={mockRemoveFromCart}
      calculateDiscountedTotal={mockCalculateDiscountedTotal}
    />,
    div
  )
})

test("ShoppingCart renders correctly", () => {
  const shoppingCart = renderer
    .create(
      <ShoppingCart
        products={mockProducts}
        shoppingCart={mockShoppingCart}
        removeFromCart={mockRemoveFromCart}
        calculateDiscountedTotal={mockCalculateDiscountedTotal}
      />
    )
    .toJSON()
  expect(shoppingCart).toMatchSnapshot()
})
