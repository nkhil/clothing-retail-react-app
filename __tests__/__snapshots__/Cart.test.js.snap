// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart renders correctly 1`] = `
<div
  id="shoppingCart"
>
  <h2>
    Basket:
  </h2>
  <ul>
    <li
      id="product1"
    >
      <h3>
        Almond Toe Court Shoes, Patent Black
         x 
        1
      </h3>
      <button
        className="remove-item"
        onClick={[Function]}
      >
        Remove
      </button>
    </li>
    <li
      id="product2"
    >
      <h3>
        Suede Shoes, Blue
         x 
        1
      </h3>
      <button
        className="remove-item"
        onClick={[Function]}
      >
        Remove
      </button>
    </li>
  </ul>
  <h3
    id="total"
  >
    Total: 
    Â£23.00
  </h3>
</div>
`;
